modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/base_uad_anonymizer.ftl

```diff
@@ -38,6 +43,17 @@ public abstract class Base${entity.name}UADAnonymizer extends DynamicQueryUADAno
 								${entity.varName}.set${uadAnonymizableEntityColumn.methodName}(anonymousUser.get${textFormatter.format(uadAnonymizableEntityColumn.UADAnonymizeFieldName, 6)}());
 							</#list>
 						</#if>
+
+						<#if hasAssetEntry && stringUtil.equals(uadUserIdEntityColumn.name, "userId")>
+							AssetEntry assetEntry = assetEntryLocalService.fetchEntry(${entity.name}.class.getName(), ${entity.varName}.get${entity.getPKMethodName()}());
+
+							if (assetEntry != null) {
+								assetEntry.setUserId(anonymousUser.getUserId());
+								assetEntry.setUserName(anonymousUser.getFullName());
+
+								assetEntryLocalService.updateAssetEntry(assetEntry);
+							}
+						</#if>
```

Comment:
We can extract this to a separate method `protected updateAsset(Object entity, long userId, User anonyUser)` so it'll be easier to override if an entity requires special logic to update its assets.

Actually, we can break it down even further to have another method for getting the asset entry `protected getAssetEntry(Object entity)`. So for cases like the WikiPage, we only need to override this simple method.